### PR TITLE
ref(images-loaded): Add candidate list filter

### PR DESCRIPTION
**Pull request by priscilawebdev**
Preview:
image

images-loaded-filter

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidate/utils.tsx [hunk 1]
```diff
@@ -64,3 +69,72 @@ export function getSourceTooltipDescription(
...
+        label: (
+          <React.Fragment>
+            {t('No debug file was not found at this location')}
+            {':'}
+          </React.Fragment>
+        ),
+        description: (
+          <React.Fragment>
+            <div>{location}</div>
+            {details && <div>{details}</div>}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidate/utils.tsx:106]: Should these divs be `p` element so you get margins between each line?

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidate/utils.tsx [hunk 2]
```diff
@@ -64,3 +69,72 @@ export function getSourceTooltipDescription(
...
+      return {
+        label: t(
+          'This issue was processed before this debug information file was available. To apply new debug information, reprocess this issue. '
+        ),
+      };
+    }
+    default: {
+      Sentry.withScope(scope => {
+        scope.setLevel(Sentry.Severity.Warning);
+        Sentry.captureException(new Error("Unknown Image's candidate download status"));
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidate/utils.tsx:135]:
```suggestion
        Sentry.captureException(new Error("Unknown image candidate download status"));
```

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx [hunk 3]
```diff
@@ -22,69 +34,379 @@ type Props = {
...
+      this.setState({
+        filteredCandidatesBySearch: candidates,
+        filteredCandidatesByFilter,
+      });
+      return;
+    }
+
+    // Slightly hacky, but it works
+    // the string is being `stringfy`d here in order to match exactly the same `stringfy`d string of the loop
+    const searchFor = JSON.stringify(searchTerm)
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx:86]: Do we have enough common behavior between this and Breadcrumbs that we could have a generalized component for this dropdown + search pattern?

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx:86]: yes I copied it from the breadcrumbs :D and thought also about creating a common component. I will do that in a follow-up PR 😉

**Review by markstory — APPROVED** [review]
Looks good code wise.

**Pull request merged**